GoMoney/ViewModel/Detail/DetailViewModel.swift

```diff
@@ -0,0 +1,30 @@
+import Realm
+import UIKit
+
+class DetailViewModel {
+    var transaction: Expense?
+    private let service = DataService.shared
+
+    func applyTransaction(newTrans: Expense, completion: ((Error?) -> Void)? = nil) {
+        guard let transaction = transaction else {
+            completion?(DataError.noTransactions)
+            return
+        }
+        DataService.shared.updateExpense(
+            oldTrans: transaction,
+            newTrans: newTrans,
+            completion: { _ in
+                completion?(nil)
+            })
```

Comment:
```suggestion
        DataService.shared.updateExpense(
            oldTrans: transaction,
            newTrans: newTrans,
            completion: { error in
                completion?(error)
            })
```
because if alway completion with `nil` -> alway success => can't handle error if update fail from database